File: COS20007/3.2P/3.2P/Clock.cs
using System;
using System.Timers;

namespace CounterObject
{
    public class Clock
    {
        private Counter _hour;
        private Counter _minute;
        private Counter _second;
        private Timer _tick;

        public Clock()
        {
            _hour = new Counter("Hour");
            _minute = new Counter("Minute");
            _second = new Counter("Second");

            _tick = new Timer();
            _tick.Interval = 100;
            _tick.Enabled = true;
            _tick.Elapsed += _tick_Elapsed;
            _tick.Start();
        }

        private void _tick_Elapsed(object sender, ElapsedEventArgs e)
        {
            Tick();
        }

        public void Tick()
        {
            if (_second.Value < 60)
            {
                _second.Increment();
            }
            else
            {
                _second.Reset();
                if (_minute.Value < 60)
                {
                    _minute.Increment();
                }
                else
                {
                    _minute.Reset();
                    if (_hour.Value < 24)
                    {
                        _hour.Increment();
                    }
                    else
                    {
                        _hour.Reset();
                    }
                }

            }
        }

        public string Time
        {
            get
            {
                return (_hour.Value).ToString("00") + ":" + (_minute.Value).ToString("00") + ":" + (_second.Value).ToString("00");
            }
        }

        public void Reset()
        {
            _hour.Reset();
            _minute.Reset();
            _second.Reset();
        }
    }
}

File: COS20007/10.1P/Clock.ts
import Counter from './Counter';

export default class Clock {
  private _hour: Counter;
  private _minute: Counter;
  private _second: Counter;

  public constructor() {
    this._hour = new Counter('Hour');
    this._minute = new Counter('Minute');
    this._second = new Counter('Second');
  }

  public Tick(): void {
    if (this._second.Value < 60) {
      this._second.Increment();
    } else {
      this._second.Reset();
      if (this._minute.Value < 60) {
        this._minute.Increment();
      } else {
        this._minute.Reset();
        if (this._hour.Value < 24) {
          this._hour.Increment();
        } else {
          this._hour.Reset();
        }
      }
    }
  }

  public get Time(): string {
    const pad = (num: number) => {
      let value = num.toString();
      return value.length == 1 ? `0${value}` : value;
    }
    return `${pad(this._hour.Value)}:${pad(this._minute.Value)}:${pad(this._second.Value)}`;
  }

  public Reset(): void {
    this._hour.Reset();
    this._minute.Reset();
    this._second.Reset();
  }
}